src/cmake-tools.ts

```diff
@@ -319,18 +319,12 @@ export class CMakeTools implements vscode.Disposable, api.CMakeToolsAPI {
             const relPathDir: string = lightNormalizePath(path.relative(this.folder.uri.fsPath, fullPathDir));
             const joinedPath = "${workspaceFolder}/".concat(relPathDir);
             vscode.workspace.getConfiguration('cmake', this.folder.uri).update("sourceDirectory", joinedPath);
-            const drv = await this.getCMakeDriverInstance();
-            if (drv) {
-              drv.config.updatePartial({sourceDirectory: joinedPath});
-            } else {
-              const reloadWindowButton = localize('reload.window', 'Reload Window');
-              const reload = await vscode.window.showErrorMessage(localize('setting.sourceDirectory.failed.needs.reload.window',
-                    'Something went wrong while updating the cmake.sourceDirectory setting. Please run the "Reload window" command for the change to take effect.'),
-                    reloadWindowButton);
-              if (reload === reloadWindowButton) {
-                vscode.commands.executeCommand('workbench.action.reloadWindow');
+            // tslint:disable-next-line: no-floating-promises
+            this.getCMakeDriverInstance().then (d => {
```

Comment:
`cmakePreConditionProblemHandler` is invoked by CMakeDriver (in `_beforeConfigureOrBuild`).  It would be nice if you could just pass in the `config` or the `driver` directly to avoid making this call.

Reply:
I tried but config.updatePartial generates another configure during this configure, causing error popup. I wasn't able to quickly rewire everything for this not to happen. It looks like the ".then" ensures updatePartial happens at a better timing. Do you think I overlook something?

Reply:
Can you add an optional parameter to `updatePartial` that instruct it not to fire the events that trigger a new configure?  The configure will happen after this function returns so it should be safe to disable those events.  (you can correct me if I'm wrong though)